File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_Billboards.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_Billboard import	t_Billboard
#/==========================================================================
#*!
#	@brief	t_Billboards
#/
class t_Billboards:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	aBillboard = []	#t_Billboard
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.aBillboard = []

	def read(self,cVariable):
		n = cVariable.getU32()
		if (n > 65536) :
			cVariable.error('array size error in aBillboard')
			return False
		self.aBillboard = [None]*n
		for i in range(n):
			self.aBillboard[i] = t_Billboard()
			if (not self.aBillboard[i].read(cVariable)):
				return False
		return True

	def write(self,cVariable):
		if (len(self.aBillboard) > 65536) :
			cVariable.error('array size error in aBillboard')
			return False
		n = len(self.aBillboard)
		cVariable.putU32(n)
		for i in range(n):
			if (not self.aBillboard[i].write(cVariable)):
				return False
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_Billboards.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_Billboards
*/
public class t_Billboards : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public t_Billboard[]	aBillboard;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_Billboards() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		aBillboard = new t_Billboard[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			{
				int n = (int) cVariable.getU32();
				if (n > 65536) {
					cVariable.error("array size error in aBillboard");
					return false;
				}
				aBillboard = new t_Billboard [n];
				for(int i = 0;i < n;++i) {
					aBillboard[i] = new t_Billboard();
					if (!aBillboard[i].read(cVariable)) return false;
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		{	
				if (aBillboard.Length > 65536) {
					cVariable.error("array size error in aBillboard");
					return false;
				}
			int n = aBillboard.Length;
			cVariable.put((uint) n);
			for(int i = 0;i < n;++i) {
				if (!aBillboard[i].write(cVariable)) return false;

			}
		}
		return true;
	}
};

